components/patient-measurements/api/src/main/java/org/phenotips/measurements/internal/AbstractMeasurementHandler.java

```diff
@@ -166,6 +199,51 @@ public void initialize() throws InitializationException
         this.chartConfigurations = this.settingsFactory.loadConfigurationsForMeasurementType(getName());
     }
 
+    @Override
+    public Collection<VocabularyTerm> getAssociatedTerms(Double standardDeviation)
+    {
+        ResourceBundle configuration = ResourceBundle.getBundle("measurementsAssociatedTerms");
+        List<VocabularyTerm> terms = new ArrayList<>();
+
+        if (null == standardDeviation || standardDeviation <= -3) {
+            addResolvedTermsToList(configuration, this.getName(), "extremeBelowNormal", terms);
+        }
+        if (null == standardDeviation || standardDeviation <= -2) {
```

Comment:
The repeating null checks are there on purpose -- if null is passed, *all* terms are supposed to be returned. Setting to negative or positive infinity will not accomplish that because the positive SD terms will be missed.

`addResolvedTermsToList()` can't be factored out because, again, there might be multiple terms to add to the list in one method call. This is also why I used `if` instead of `else if`.

I'm not using the constants in `MeasurementUtils` because those ones are `delimiter-separated` as opposed to `camelCase`.